src/storage/storage.service.ts

```diff
@@ -0,0 +1,82 @@
+import { Injectable, NotFoundException } from '@nestjs/common';
+import { ConfigService } from '@nestjs/config';
+import {
+    createReadStream,
+    createWriteStream,
+    promises as fsPromises,
+    write,
+} from 'fs';
+import { join } from 'path';
+import { Stream } from 'stream';
+import { promisify } from 'util';
+import { v4 as uuid } from 'uuid';
+
+const pipeline = promisify(Stream.pipeline);
+
+@Injectable()
+export class StorageService {
+    private readonly storage: string | undefined;
+    private readonly html = 'html';
+    private readonly xml = 'xml';
+
+    constructor(private readonly configService: ConfigService) {
+        this.storage = this.configService.get<string>('STORAGE');
+        if (!this.storage) {
+            throw new Error('Something wrong with env or config service');
+        }
+    }
+
+    saveFile = async (file: Express.Multer.File): Promise<string> => {
+        const fileId: string = uuid();
+        const fileExtension: string = file.mimetype.includes(this.html)
+            ? this.html
+            : this.xml;
+
+        const filePath = join(
+            process.cwd(),
+            `${this.storage}/${fileId}.${fileExtension}`,
```

Comment:
ja ta kropke bym dodal do rozszerzenia juz wyzej

Reply:
ten slash tez jest zle dodany bo na windowsie to nie zadziala chyba bo sa w odwrotna strone jak sie nie myle... sprawdzcie czy nie ma przypadekim w node.js jakies stalej ktora definiuje slash w zaleznosci od systemu 

przyklad jak to jest w Javie zrobione 
https://docs.oracle.com/javase/8/docs/api/java/io/File.html#separator

Reply:
path.join() działa podając stringi po przecinku, nie trzeba robić slasha - właśnie po to jest path.join (różne systemy):
```
const filePath = join(
            process.cwd(),
            this.storage,
            `${fileId}.${fileExtension}`
```
?

Reply:
tak tylko ze @Adam-DevPL wstawia slasha i dlatego mowie ze to nie jest dobre bo to na windowsach nie zadziala bo one maja w 2 strone slasha

Reply:
do poprawy